lib/box/file.ts

```diff
@@ -3,16 +3,38 @@ import { readFile, readFileSync, stat, statSync, type ReadFileOptions } from 'he
 import type fs from 'fs';
 
 class File {
+
+  /**
+   * Full path of the file
+   */
   public source: string;
+
+  /**
+   * Relative path to the box of the file
+   */
   public path: string;
+
+  /**
+   * The information from path matching.
+   */
   public params: any;
-  public type: string;
+
+  /**
+   * File type. The value can be create, update, skip, delete.
+   */
+  // eslint-disable-next-line no-use-before-define
+  public type: typeof File.TYPE_CREATE | typeof File.TYPE_UPDATE | typeof File.TYPE_SKIP | typeof File.TYPE_DELETE;
   static TYPE_CREATE: 'create';
   static TYPE_UPDATE: 'update';
   static TYPE_SKIP: 'skip';
   static TYPE_DELETE: 'delete';
 
-  constructor({ source, path, params, type }) {
+  constructor({ source, path, params, type }: {
+    source: string;
+    path: string;
+    params: any;
+    type: typeof File.TYPE_CREATE | typeof File.TYPE_UPDATE | typeof File.TYPE_SKIP | typeof File.TYPE_DELETE;
```

Comment:
```suggestion
    type: typeof this['type'];
```

Haven't try this, but it should work and simplify things a little.

Reply:
oops
> TS2526: A 'this' type is available only in a non-static member of a class or interface.